components/spiffs/include/spiffs_config.h

```diff
@@ -158,7 +158,8 @@ extern void spiffs_api_unlock(struct spiffs_t *fs);
 // This is derived from following:
 // logical_page_size - (SPIFFS_OBJ_NAME_LEN + sizeof(spiffs_page_header) +
 // spiffs_object_ix_header fields + at least some LUT entries)
-#define SPIFFS_OBJ_META_LEN             (CONFIG_SPIFFS_META_LENGTH)
+// We are using at least 1 byte of metadata (for directory handling)
+#define SPIFFS_OBJ_META_LEN             (CONFIG_SPIFFS_META_LENGTH+1)
```

Comment:
This change makes it impossible to set the length of metadata to zero. SPIFFS_OBJ_META_LEN must be equal to the value set in Kconfig.